### PR TITLE
test: superchain rev calc unit test

### PR DESCRIPTION
**Pull request by funkornaut001**
Closes OPT-1073
Closes OPT-1076

### CODE REVIEW

File: packages/contracts-bedrock/test/L2/SuperchainRevSharesCalculator.t.sol [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+
+/// @notice Base setup contract for SuperchainRevSharesCalculator tests.
+contract SuperchainRevSharesCalculator_TestInit is CommonTest {
+    ISuperchainRevSharesCalculator calculator;
+    address payable shareRecipient;
+    address payable remainderRecipient;
+
+    event ShareRecipientUpdated(address indexed newShareRecipient, address indexed oldShareRecipient);
+    event RemainderRecipientUpdated(address indexed newRemainderRecipient, address indexed oldRemainderRecipient);
```
Comment (0xDiscotech) [inline packages/contracts-bedrock/test/L2/SuperchainRevSharesCalculator.t.sol]: Normally, this goes in the inverse order (OZ does it in this way). We should also be consistent with the `FeeSplitter`. Could you add this change on this PR?
```suggestion
    event ShareRecipientUpdated(address indexed oldShareRecipient, address indexed newShareRecipient);
    event RemainderRecipientUpdated(address indexed oldRemainderRecipient, address indexed newRemainderRecipient);
```

File: packages/contracts-bedrock/test/L2/SuperchainRevSharesCalculator.t.sol [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+        vm.assume(_caller != proxyAdminOwner);
+        
+        vm.expectRevert(ISuperchainRevSharesCalculator.SharesCalculator_OnlyProxyAdminOwner.selector);
+        vm.prank(_caller);
+        calculator.setRemainderRecipient(_newRemainderRecipient);
+    }
+
+    /// @notice Tests that setRemainderRecipient updates recipient and emits event.
+    function testFuzz_setRemainderRecipient_succeeds(address payable _newRemainderRecipient) external {
+
```
Comment (0xDiscotech) [inline packages/contracts-bedrock/test/L2/SuperchainRevSharesCalculator.t.sol]:
```suggestion
```

File: packages/contracts-bedrock/test/L2/SuperchainRevSharesCalculator.t.sol [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+        
+        assertEq(address(calculator.remainderRecipient()), address(_newRemainderRecipient));
+    }
+}
+
+/// @notice Tests for SuperchainRevSharesCalculator getRecipientsAndAmounts function.
+contract SuperchainRevSharesCalculator_getRecipientsAndAmounts_Test is SuperchainRevSharesCalculator_TestInit {
+    
+    /// @notice Test that getRecipientsAndAmounts reverts when gross share is calculated to be 0.
+    function test_getRecipientsAndAmounts_zeroGrossShare_reverts() external {
```
Comment (0xDiscotech) [inline packages/contracts-bedrock/test/L2/SuperchainRevSharesCalculator.t.sol]: Instead of hardcoding `0`, could we bound an amount that will go from 0 to the max value that will end up in 0 while doing the math as well?

File: packages/contracts-bedrock/test/L2/SuperchainRevSharesCalculator.t.sol [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+    function testFuzz_getRecipientsAndAmounts_grossShareHigher_succeeds(
+        uint256 _sequencerFees,
+        uint256 _baseFees,
+        uint256 _operatorFees,
+        uint256 _l1Fees
+    ) external view {
+        // Use smaller bounds to prevent overflow
+        _sequencerFees = bound(_sequencerFees, 1, type(uint64).max);
+        _baseFees = bound(_baseFees, 1, type(uint64).max);
+        _operatorFees = bound(_operatorFees, 1, type(uint64).max);
```
Comment (0xDiscotech) [inline packages/contracts-bedrock/test/L2/SuperchainRevSharesCalculator.t.sol]: I know that is to avoid an overflow, however, we could be omitting an edge case here since the max of `uint64` is 18 ethers. I suggest changing to `uint80`, which I think we'll be good, or, handling it differently the bounds
```suggestion
        _sequencerFees = bound(_sequencerFees, 1, type(uint80).max);
        _baseFees = bound(_baseFees, 1, type(uint80).max);
        _operatorFees = bound(_operatorFees, 1, type(uint80).max);
```

Comment (0xDiscotech) [reply packages/contracts-bedrock/test/L2/SuperchainRevSharesCalculator.t.sol]: The same can be applied on the other tests were this patterns repeats

**Review by 0xDiscotech — COMMENTED** [review]
Great job so far! Left few comments

File: packages/contracts-bedrock/test/L2/SuperchainRevSharesCalculator.t.sol (hunk 1 above)
Comment (funkornaut001) [reply packages/contracts-bedrock/test/L2/SuperchainRevSharesCalculator.t.sol]: https://github.com/defi-wonderland/optimism/pull/520/commits/6f5a7c5ac49c456498b71232273e357b09860f7d

File: packages/contracts-bedrock/test/L2/SuperchainRevSharesCalculator.t.sol (hunk 4 above)
Comment (funkornaut001) [reply packages/contracts-bedrock/test/L2/SuperchainRevSharesCalculator.t.sol]: Uint112 is working :) https://github.com/defi-wonderland/optimism/pull/520/commits/4d8413333ed6b1336ccd6aaafe3cd1b3cd3c863f

File: packages/contracts-bedrock/test/L2/SuperchainRevSharesCalculator.t.sol (hunk 3 above)
Comment (funkornaut001) [reply packages/contracts-bedrock/test/L2/SuperchainRevSharesCalculator.t.sol]: https://github.com/defi-wonderland/optimism/pull/520/commits/830b0eceebc1f6cf05060394ab097017b68aa9b0

Comment (0xDiscotech) [reply packages/contracts-bedrock/test/L2/SuperchainRevSharesCalculator.t.sol]: Perfect!
